auctions/english-auction/src/data_structures.sw

```diff
@@ -0,0 +1,197 @@
+library data_structures;
+
+dep errors;
+
+use errors::AssetError;
+use std::{
+    assert::require,
+    contract_id::ContractId,
+    identity::Identity,
+    option::Option,
+    storage::StorageMap,
+};
+
+pub enum Asset {
+    NFTAsset: NFTAsset,
+    TokenAsset: TokenAsset,
+}
+
+pub enum State {
+    Closed: (),
+    Open: (),
+}
+
+pub struct Auction {
```

Comment:
Consistency is key. I've mentioned this recently to @matt-user . It makes it easier to review because I don't have to figure out how you structure your code. I just need to know the alphabet. I think it makes sense to keep the trait impl's below the declaration too